core/src/main/java/io/grpc/transport/AbstractStream.java

```diff
@@ -42,7 +44,13 @@
 /**
  * Abstract base class for {@link Stream} implementations.
  */
-public abstract class AbstractStream<IdT> implements Stream {
+public abstract class AbstractStream<IdT, LT extends StreamListener> implements Stream {
+  /**
+   * The default number of queued bytes for a given stream, below which
+   * {@link StreamListener#onReady(int)} will be called.
+   */
+  public static final int DEFAULT_ONREADY_THRESHOLD = 1048576;
```

Comment:
This can be a much lower number, since the bandwidth-delay product between the application and the transport is the _network's_ bandwidth * context switch latency.

Netty [uses 32K for instance](https://github.com/netty/netty/blob/8f8a06ab0a885accf72b19cb27f214072204726b/transport/src/main/java/io/netty/channel/DefaultChannelConfig.java#L62).